File: src/test/java/no/nav/syfo/localconfig/LocalApplicationConfig.java
package no.nav.syfo.localconfig;

import no.nav.security.spring.oidc.test.TokenGeneratorConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import org.springframework.core.env.Environment;

import static java.util.Objects.requireNonNull;

@Configuration
@Import(TokenGeneratorConfiguration.class)
public class LocalApplicationConfig {

    public LocalApplicationConfig(Environment environment) {
        System.setProperty("SECURITYTOKENSERVICE_URL", requireNonNull(environment.getProperty("securitytokenservice.url")));
        System.setProperty("SRVMODIASYFOREST_USERNAME", requireNonNull(environment.getProperty("srvmodiasyforest.username")));
        System.setProperty("SRVMODIASYFOREST_PASSWORD", requireNonNull(environment.getProperty("srvmodiasyforest.password")));
    }
}

File: src/test/kotlin/no/nav/syfo/LocalApplicationConfig.kt
package no.nav.syfo

import no.nav.security.oidc.test.support.spring.TokenGeneratorConfiguration
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import org.springframework.core.env.Environment
import java.util.*

@Configuration
@Import(TokenGeneratorConfiguration::class)
class LocalApplicationConfig(environment: Environment) {
    init {
        System.setProperty("SECURITYTOKENSERVICE_URL", Objects.requireNonNull(environment.getProperty("securitytokenservice.url")))
        System.setProperty("SRVMODIASYFOREST_USERNAME", Objects.requireNonNull(environment.getProperty("srvmodiasyforest.username")))
        System.setProperty("SRVMODIASYFOREST_PASSWORD", Objects.requireNonNull(environment.getProperty("srvmodiasyforest.password")))
    }
}
